releases/release-4.0.10.md

```diff
@@ -0,0 +1,110 @@
+    - Fix a bug that the schema change causes incorrect results [#21596](https://github.com/pingcap/tidb/pull/21596)
+    - Avoid unnecessary column flag changes incurred by `ALTER TABLE` [#21474](https://github.com/pingcap/tidb/pull/21474)
+    - Set the database name for table aliases of query blocks used in optimizer hints [#21380](https://github.com/pingcap/tidb/pull/21380)
+    - Generate the proper hint for `IndexHashJoin`/`IndexMergeJoin` [#21020](https://github.com/pingcap/tidb/pull/21020)
+
++ TiKV
+
+    - Fix the wrong mapping between ready and peer [#9409](https://github.com/tikv/tikv/pull/9409)
+    - Fix the issue that some logs are not redacted when `security.redact-info-log` is set to `true` [#9314](https://github.com/tikv/tikv/pull/9314)
+
++ PD
+
+    - Fix the issue that the ID allocation is not monotonic [#3308](https://github.com/tikv/pd/pull/3308) [#3323](https://github.com/tikv/pd/pull/3323)
+    - Fix the issue that the PD client might be blocked in some cases [#3285](https://github.com/pingcap/pd/pull/3285)
+
++ TiFlash
+
+    - Fix the issue that TiFlash fails to start because TiFlash fails to process the TiDB schema with an old version
+    - Fix the issue that TiFlash fails to start due to incorrect handling of `cpu_time` on the RedHat system
+    - Fix the issue that TiFlash fails to start when `path_realtime_mode` is `true`
+    - Fix an issue of incorrect results when calling the `substr` function with three parameters
+    - Fix the issue that TiFlash does not support changing the `Enum` type even if the change is lossless
+
++ Tools
+
+    + TiCDC
+
+        - Fix the maxwell issues, including the `base64` data output and output TSO to unix timestamp [#1173](https://github.com/pingcap/ticdc/pull/1173)
+        - Fix a bug that outdated metadata might cause the newly created changefeed abnormal [#1184](https://github.com/pingcap/ticdc/pull/1184)
+        - Fix create receiver on closed notifier [#1199](https://github.com/pingcap/ticdc/pull/1199)
+        - Fix memory accumulated when owner consume etcd update slow [#1227](https://github.com/pingcap/ticdc/pull/1227)
+        - Fix the issue that `max-batch-size` does not take effect [#1253](https://github.com/pingcap/ticdc/pull/1253)
+        - Fix cleanup stale tasks before capture information is constructed [#1280](https://github.com/pingcap/ticdc/pull/1280)
+        - Roll back transactions to recycle db conn for MySQL sink [#1285](https://github.com/pingcap/ticdc/pull/1285)
+
+    + Dumpling
+
+        - Do not set `tidb_mem_quota_query` when it is not specified to avoid TiDB OOM [#233](https://github.com/pingcap/dumpling/pull/233)
+
+    + Backup & Restore (BR)
+
+        - Fix the issue that BR v4.0.9 cannot restore the GCS backup files of v4.0.8 [#688](https://github.com/pingcap/br/pull/688)
+        - Fix the issue that BR panics when the GCS storage URL has no prefix [#673](https://github.com/pingcap/br/pull/673)
```

Comment:
```suggestion
        - Fix the issue that BR panics when the GCS storage URL has no prefix [#673](https://github.com/pingcap/br/pull/673)
        - Disable backup stats by default to avoid BR OOM [#693](https://github.com/pingcap/br/pull/693)

```